### PR TITLE
Changes to support building on CentOS 8.

### PR DESCRIPTION
**Pull request by igorpeshansky**
- Fix the format truncation warnings.
- OpenSSL 1.1.0 compat in libmongoc.

### CODE REVIEW

File: src/collectdctl.c [hunk 1]
```diff
@@ -150,7 +150,7 @@ static int parse_identifier (lcc_connection_t *c,
     const char *value, lcc_identifier_t *ident)
 {
   char hostname[1024];
-  char ident_str[1024] = "";
+  char ident_str[1028] = "";
```
Comment (jkschulz) [inline src/collectdctl.c:153]: Does this go into fixing "the format truncation warnings"?

File: src/utils_db_query.c [hunk 2]
```diff
@@ -265,7 +265,7 @@ static int udb_result_submit (udb_result_t *r, /* {{{ */
     }
     else
     {
-      char tmp[DATA_MAX_NAME_LEN];
+      char tmp[DATA_MAX_NAME_LEN - 10];
```
Comment (jkschulz) [inline src/utils_db_query.c:268]: Why 10?

File: src/collectdctl.c (hunk 1 above)
Comment (igorpeshansky) [reply src/collectdctl.c:153]: Yeah. The error was "output 2 or more bytes (assuming 1025) into a destination of size 1024" for line 170 below, so anything over 2 bytes would have probably worked.

File: src/utils_db_query.c (hunk 2 above)
Comment (igorpeshansky) [reply src/utils_db_query.c:268]: No good reason. The error was "output 2 or more bytes (assuming 129) into a destination of size 128" in line 280 below, so anything over 2 should work. Want me to change it to 4 to match the other adjustment?

Comment (jkschulz) [reply src/utils_db_query.c:268]: it looks like DATA_MAX_NAME_LEN is defined to be 128. Is tmp[] here related to the ident_str in the other file? Would it make sense to leave this as DATA_MAX_NAME_LEN, but change the other one to use DATA_MAX_NAME_LEN too?

Comment (igorpeshansky) [reply src/utils_db_query.c:268]: No, they are unrelated. I can make the adjustment amount consistent, though...

**Review by jkschulz — APPROVED** [review]

**Pull request merged**